src/portal/mod.rs

```diff
@@ -103,48 +106,60 @@ impl Keyring {
             }
         };
 
-        let key = keyring.derive_key(secret);
-
         Ok(Self {
             keyring: RwLock::new(keyring),
             path: path.as_ref().to_path_buf(),
             mtime: RwLock::new(mtime),
-            key,
+            key: Default::default(),
+            secret,
         })
     }
 
+    /// Retrieve the number of items
+    ///
+    /// This function will not trigger a key derivation.
+    pub async fn n_items(&self) -> usize {
```

Comment:
I would be interested in having this implemented also for the dbus implementation & exposed in the "common" generic API if possible.

Reply:
Anyways, this PR only touches the `portal` part anyways so the rest can be done in a different PR.